_posts/2019-08-08-hacking-for-cruise-control.md

```diff
@@ -0,0 +1,244 @@
+
+```Dockerfile
+FROM centos:7
+
+ENV CC_VERSION=2.0.57
+ENV JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk
+
+RUN yum -y install git java-1.8.0-openjdk-devel && \
+    yum clean all -y
+
+RUN git clone --branch ${CC_VERSION} https://github.com/linkedin/cruise-control.git
+
+WORKDIR cruise-control
+
+RUN ./gradlew jar copyDependantLibs
+
+# Configure Cruise Control to point to the proper Kafka and Zookeeper endpoints
+RUN sed -i 's/bootstrap.servers=.*/bootstrap.servers=my-cluster-kafka-bootstrap:9092/g' config/cruisecontrol.properties \
+    && sed -i 's/zookeeper.connect=.*/zookeeper.connect=zoo-entrance:2181/g' config/cruisecontrol.properties \
+    && sed -i 's/capacity.config.file=.*/capacity.config.file=config\/capacityJBOD.json/g' config/cruisecontrol.properties \
+    && sed -i 's/sample.store.topic.replication.factor=.*/sample.store.topic.replication.factor=1/g' config/cruisecontrol.properties 
+
+# Ensure Cruise Control log directory is writable
+RUN mkdir logs \ 
+    && touch ./logs/kafkacruisecontrol.log \
+    && chmod a+rw -R .
+
+# Install Cruise Control GUI Frontend
+RUN curl -L https://github.com/linkedin/cruise-control-ui/releases/download/v0.1.0/cruise-control-ui.tar.gz \
+    -o /tmp/cruise-control-ui.tar.gz \
+    && tar zxvf /tmp/cruise-control-ui.tar.gz
+
+ENTRYPOINT ["/bin/bash", "-c", "./kafka-cruise-control-start.sh config/cruisecontrol.properties"]
+```
+Build the image:
+```
+docker build . -t <cruise-control-image-name>
+```
+Then, using the following file, 'cruise-control-deployment.yaml', create a Cruise Control deployment that references the Cruise Control image, 'cruise-control-image-name', and a Cruise Control service endpoint for enabling communication with Cruise Control from outside the Cruise Control pod.
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: my-cruise-control
+  labels:
+    app: cruise-control
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cruise-control
+        name: my-cruise-control
+    spec:
+      containers:
+      - name: my-cruise-control
+        image: <cruise-control-image-name>
+        imagePullPolicy: 'Always'
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: cruise-control
+    name: my-cruise-control
+  name: my-cruise-control
+spec:
+  ports:
+    - name: http-9090
+      port: 9090
+      protocol: TCP
+      targetPort: 9090
+  type: NodePort
```

Comment:
You seem to use port-forward to access the service. So does this really need to be NodePort service?